python/PyQt6/core/auto_generated/qgis.sip.in

```diff
@@ -3819,6 +3819,10 @@ times)
 
 Invalid < NULL < Values
 
+Since QGIS 4.0 the ``strictTypeCheck`` argument can be used to specify
+that variants of different types should be compared using their userType
+ID only, and not attempt to check the actual variant value.
```

Comment:
@nyalldawson  Honestly, I would have preferred a separate function like `qgsVariantTypeCompare` for the exclusive type check (the "only" in the docs).

The function doc is clear but the behavior I would have expected from the new argument is that the exact type is checked **before** checking for the value, not instead of it.

Reply:
@elpaso 
> The function doc is clear but the behavior I would have expected from the new argument is that the exact type is checked before checking for the value, not instead of it

That's exactly what happens now -- if using strict mode, we FIRST check that the types are identical. If so, we compare the actual value. If not, we compare based on the types alone.